compiler/rustc_parse_format/src/lib.rs

```diff
@@ -858,8 +858,14 @@ impl<'input> Parser<'input> {
             self.errors.insert(
                 0,
                 ParseError {
-                    description: "expected format parameter to occur after `:`".to_owned(),
-                    note: None,
+                    description: format!(
+                        "expected `{}` (alignment specifier) after `:` in format string; example: `{{:>#X5}}`",
+                        alignment
+                    ),
+                    note: Some(
+                        "alignment must be one of `<` (left), `^` (center), or `>` (right)"
+                            .to_string(),
+                    ),
```

Comment:
This note does not feel useful given that the user already wrote one of the three specifiers :thinking: